src/data/equiv/fin.lean

```diff
@@ -168,16 +168,71 @@ def sum_fin_sum_equiv : fin m ⊕ fin n ≃ fin (m + n) :=
     { dsimp, rw [dif_neg H], simp [fin.ext_iff, nat.add_sub_of_le (le_of_not_gt H)] }
   end }
 
+@[simp] lemma fin_sum_fin_equiv_apply_left (x : fin m) :
+  @fin_sum_fin_equiv m n (sum.inl x) = ⟨x.1, nat.lt_of_lt_of_le x.2 $ nat.le_add_right m n⟩ :=
+rfl
+
+@[simp] lemma fin_sum_fin_equiv_apply_right (x : fin n) :
+  @fin_sum_fin_equiv m n (sum.inr x) = ⟨m + x.1, nat.add_lt_add_left x.2 m⟩ :=
+rfl
+
+@[simp] lemma fin_sum_fin_equiv_symm_apply_left (x : fin (m + n)) (h : ↑x < m) :
+  fin_sum_fin_equiv.symm x = sum.inl ⟨x.1, h⟩ :=
+by simp [fin_sum_fin_equiv, dif_pos h]
+
+@[simp] lemma fin_sum_fin_equiv_symm_apply_right (x : fin (m + n)) (h : m ≤ ↑x) :
+  fin_sum_fin_equiv.symm x = sum.inr ⟨x.1 - m, nat.lt_of_add_lt_add_left $
+      show m + (x.1 - m) < m + n, from (nat.add_sub_of_le $ h).symm ▸ x.2⟩ :=
+by simp [fin_sum_fin_equiv, dif_neg (not_lt.mpr h)]
+
+/-- Equivalence between `fin m ⊕ fin n` and `fin (n + m)` -/
+def fin_sum_fin_equiv' : (fin m) ⊕ (fin n) ≃ fin (n + m) :=
+  equiv.trans fin_sum_fin_equiv (fin_congr (add_comm m n))
```

Comment:
If this really useful as a separate definition? I'm dubious.

Reply:
I'd also be inclined to remove it. Are we missing a simp lemma about `fin_congr` that made this desirable?

Reply:
It certainly helps for the application I have in mind, but we can wait to discuss further when I PR the application...

Reply:
If you want to save on `equiv.trans _ (fin_congr _)`, you can do what @sgouezel did in some code related to `multilinear_map` and analytic functions:
```
def fin_sum_fin_equiv (m n k : nat) (h : m + n = k) : (fin m) ⊕ (fin n) ≃ fin k :=
```
If you want `fin_sum_fin_equiv m n` to work, you can try `h : m + n = k . try_refl_tac`.